### PR TITLE
[FLINK-25565][Formats][Parquet] write and read parquet int64 timestamp

### PR DESCRIPTION
**Pull request by tweise**
backport #18304

### CODE REVIEW

**Review by mbalassi — APPROVED** [review]
Approving it based on https://github.com/apache/flink/pull/18304, as the diff is the same. Thanks for the backport, @tweise.

File: flink-formats/flink-parquet/src/main/java/org/apache/flink/formats/parquet/vector/reader/TimestampColumnReader.java [hunk 1]
```diff
@@ -109,6 +157,37 @@ public static TimestampData int96ToTimestamp(
...
+            case MICROS:
+                milliseconds = value / MICROS_PER_MILLISECOND;
+                nanosOfMillisecond = (value % MICROS_PER_SECOND) * NANOS_PER_MICROSECONDS;
+                break;
+            case NANOS:
+                milliseconds = value / NANOS_PER_MILLISECOND;
+                nanosOfMillisecond = value % NANOS_PER_SECOND;
+                break;
+            default:
+                break;
```
Comment (mxm) [inline flink-formats/flink-parquet/src/main/java/org/apache/flink/formats/parquet/vector/reader/TimestampColumnReader.java:179]: No time unit means drop the value? Should we throw an error?

Comment (tweise) [reply flink-formats/flink-parquet/src/main/java/org/apache/flink/formats/parquet/vector/reader/TimestampColumnReader.java:179]: Good catch. LogicalTypeAnnotation.TimeUnit has only the 3 values that are all covered, so there should neither be a practical danger of dropping the value nor big risk of throwing an exception under default for clarity. It is also more of a stylistic concern that we can deal with in the master branch. I will open a PR there.

Comment (mxm) [reply flink-formats/flink-parquet/src/main/java/org/apache/flink/formats/parquet/vector/reader/TimestampColumnReader.java:179]: I would throw an exception for defensive programming.

**Pull request merged**